backends/hip-ref/ceed-hip-ref-operator.c

```diff
@@ -1906,9 +1906,19 @@ static int CeedOperatorLinearAssembleAddDiagonalAtPoints_Hip(CeedOperator op, Ce
       if (!is_active) continue;
 
       // Update unit vector
-      if (s == 0) CeedCallBackend(CeedVectorSetValue(active_e_vec_in, 0.0));
-      else CeedCallBackend(CeedVectorSetValueStrided(active_e_vec_in, s - 1, e_vec_size, 0.0));
-      CeedCallBackend(CeedVectorSetValueStrided(active_e_vec_in, s, e_vec_size, 1.0));
+      {
+        CeedInt  node = (s - 1) % elem_size, comp = (s - 1) / elem_size;
+        CeedSize start = node * 1 + comp * (elem_size * num_elem);
+        CeedSize stop  = start + (num_elem - 1) * elem_size + 1;
+
+        if (s == 0) CeedCallBackend(CeedVectorSetValue(active_e_vec_in, 0.0));
+        else CeedCallBackend(CeedVectorSetValueStrided(active_e_vec_in, start, stop, elem_size, 0.0));
+
+        node = s % elem_size, comp = s / elem_size;
+        start = node * 1 + comp * (elem_size * num_elem);
+        stop  = start + (num_elem - 1) * elem_size + 1;
```

Comment:
Can this expression be simpler, perhaps like
```suggestion
        stop  = (comp + 1) * elem_size * num_elem;
```

Reply:
it would need to be `stop = node + (comp + 1) * (elem_size * num_elem) - elem_size + 1` and I don't know if that's much cleaner?

Reply:
Seems to me the bound doesn't need to be sharp, just prevent a full extra stride.

Reply:
Agreed, but if we overshot then we can give a stop past the length of the array. I previously had a less sharp bound and it errored in some tests

Reply:
I think that doesn't happen with the expression I gave (one past the end of the array is still allowed).

Reply:
I think I just derived something similar to yours by substituting in (elem_size -1) for node

Reply:
fixed